### PR TITLE
Create C++ example dataflow

### PR DESCRIPTION
**Pull request by phil-opp**
This example shows how to create dora operators and custom nodes with C++.

Dora does not provide a C++ API yet, but we can create adapters for either the C or Rust API. The `operator-rust-api` and `node-rust-api` folders implement an example operator and node based on dora's Rust API, using the `cxx` crate for bridging. The `operator-c-api` and `node-c-api` show how to create operators and nodes based on dora's C API. Both approaches work well, so you can choose the API that fits your application better.

### CODE REVIEW

File: examples/c++-dataflow/operator-rust-api/src/lib.rs [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+        let mut output_sender = OutputSender(output_sender);
+
+        let result = ffi::on_input(operator, id, data, &mut output_sender);
+        if result.error.is_empty() {
+            Ok(match result.stop {
+                false => DoraStatus::Continue,
+                true => DoraStatus::Stop,
+            })
+        } else {
+            Err(())
```
Comment (haixuanTao) [inline examples/c++-dataflow/operator-rust-api/src/lib.rs:74]: Shouldn't the error message be not empty. And so the error should be: `Err(result.error)`?

**Review by haixuanTao — APPROVED** [review]
Awesome example! Thanks!

File: examples/c++-dataflow/operator-rust-api/src/lib.rs (hunk 1 above)
Comment (phil-opp) [reply examples/c++-dataflow/operator-rust-api/src/lib.rs:74]: Right now the operator API only supports an integer return code to signal a status. So we can only set an error status to signal that _something_ has gone wrong, but cannot pass an error message right now.

Comment (haixuanTao) [reply examples/c++-dataflow/operator-rust-api/src/lib.rs:74]: I see. Make sense.Thanks !

**Pull request merged**